.github/workflows/WhenMainChange.yml

```diff
@@ -0,0 +1,55 @@
+name: Build Android
+
+on:
+  push:
+    branches: [ "main" ]
+  workflow_dispatch:
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin' # See 'Supported distributions' for available options
+          java-version: '17'
+          cache: gradle
+
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
+      - name: Build with Gradle
+        run: ./gradlew :app:assembleRelease
+      - run: |
+          
+          rootDir="./" 
+          GRADLE_FILE=$rootDir"app/build.gradle.kts"
+          fileQA=$rootDir"app/build/outputs/apk/release/app-release-unsigned.apk"
+          name=Esmorga
+          versionName=$(grep "versionName" $GRADLE_FILE | awk '{print $3}')
+          versionCode=$(grep "versionCode" $GRADLE_FILE | awk '{print $3}')
+          buildNumber=""
+          buildHash=$(md5sum $fileQA | awk '{print $4}')
+          changelog=""
```

Comment:
Do we need to set an empty val?

Reply:
No, I am simply following the original schema that Andres provided to me, as I believe it will be useful in the future if we wish to add change log data without having to recheck how OtaShare functions. However, it can be removed if it is not being utilized.

Reply:
if it's not used, it shouldn't be in the code.

Reply:
ok, removing it.